### PR TITLE
Fix build with ALTSVC enabled and COOKIES disabled

### PR DESCRIPTION
**Pull request by sunpoet**
ALTSVC requires Curl_get_line which is defined in lib/cookie.c inside a #if
check of HTTP and COOKIES. That makes Curl_get_line undefined if COOKIES is
disabled. This is a workaround to define Curl_get_line unconditionally.

### CODE REVIEW

**Comment by sunpoet** [comment]
This is just the workaround [I committed for FreeBSD ports tree](https://svnweb.freebsd.org/changeset/ports/497348). You might want to move it to a new place, e.g. lib/curl_get_line.c and lib/curl_get_line.h.

**Comment by bagder** [comment]
I agree with you @danielgustafsson. I was mostly just lazy when I left it there in my alt-svc patch...

**Comment by danielgustafsson** [comment]
> I agree with you @danielgustafsson. I was mostly just lazy when I left it there in my alt-svc patch...

Cool, let's do it. @sunpoet do you want to pursue this patch? If not then I'm fine to pick it up.

**Comment by sunpoet** [comment]
> > I agree with you @danielgustafsson. I was mostly just lazy when I left it there in my alt-svc patch...
> 
> Cool, let's do it. @sunpoet do you want to pursue this patch? If not then I'm fine to pick it up.

I've prepared the patch which moves Curl_get_line to lib/curl_get_line.[ch].

File: lib/cookie.h [hunk 1]
```diff
@@ -24,6 +24,7 @@
 #include "curl_setup.h"
 
 #include <curl/curl.h>
+#include "curl_get_line.h"
```
Comment (MarcelRaad) [inline lib/cookie.h]: This isn't used in the header, is it? I'd rather move it to the implementation file.

File: lib/curl_get_line.c [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+ * furnished to do so, under the terms of the COPYING file.
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTY OF ANY
+ * KIND, either express or implied.
+ *
+ ***************************************************************************/
+
+#include "curl_setup.h"
+
+#include "curl_get_line.h"
```
Comment (danielgustafsson) [inline lib/curl_get_line.c]: This should also include the following headers *after* `curl_get_line.h`:
```c
#include "curl_memory.h"
/* The last #include file should be: */
#include "memdebug.h"
```

**Comment by bagder** [comment]
@sunpoet, are you up to taking this all the way to the finishing line?

**Comment by sunpoet** [comment]
I just committed the changes suggested by @MarcelRaad and @danielgustafsson. Thanks!

**Comment by danielgustafsson** [comment]
Pushed squashed and with some minor editorialization to the commit message, thanks for contributing!

**Comment by sunpoet** [comment]
Thanks!

**Pull request closed without merging**